### PR TITLE
Add ZodErrorMap

### LINKED ISSUE #91: Zod errors in config are not always very helpful
### What version of `starlight` are you using?

0.0.12

### What version of `astro` are you using?

2.5.0

### What package manager are you using?

pnpm

### What operating system are you using?

macOS / Stackblitz

### What browser are you using?

Firefox

### Describe the Bug

When a user provides invalid configuration options to the Starlight integration these are caught by the Zod schema and an error is displayed.

For example the `title` option is required. If omitted, the following is logged:

```
[vite] Error when evaluating SSR module /home/projects/github-qyfkex/astro.config.mjs:
|- Error
    at ZodError (file:///home/projects/github-qyfkex/node_modules/zod/lib/index.mjs:283:9)
    at get error (file:///home/projects/github-qyfkex/node_modules/zod/lib/index.mjs:656:31)
    at parse (file:///home/projects/github-qyfkex/node_modules/zod/lib/index.mjs:755:9)

[astro] Unable to load /home/projects/github-qyfkex/astro.config.mjs

 error   Required

```

This message does not mention that the `title` field is what is missing/required, nor that the error is in the Starlight config. It would be good to improve this!

@bholmesdev suggested we can improve this using a Zod “error map”. Here is the content collections one for reference: https://github.com/withastro/astro/blob/main/packages/astro/src/content/error-map.ts and there’s more detail about [error handling in Zod’s docs](https://github.com/colinhacks/zod/blob/master/ERROR_HANDLING.md). We parse user config in the main integration entrypoint: https://github.com/withastro/starlight/blob/fe1c489a41a48eceed3cf559374d07d67c325248/packages/starlight/index.ts#L22

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-qyfkex?file=astro.config.mjs

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by TheOtterlord**
#### What kind of changes does this PR include?

- Changes to Starlight code

#### Description

- Closes #91 
- What does this PR change? Adds an error map for zod errors ([modified from astro content collections](https://github.com/withastro/astro/blob/main/packages/astro/src/content/error-map.ts))

```
15:23:05 [astro] Unable to load /home/otterlord/Documents/Repos/starlight/docs/astro.config.mjs

 error   **logo**: Did not match union:
  > **logo.replacesTitle**: Expected type `"boolean"`, received "string"
```

### CODE REVIEW

File: packages/starlight/index.ts [hunk 1]
```diff
@@ -15,11 +15,20 @@ import {
...
 export default function StarlightIntegration(
   opts: StarlightUserConfig
 ): AstroIntegration[] {
-  const userConfig = StarlightConfigSchema.parse(opts);
+  const parsedConfig = StarlightConfigSchema.safeParse(opts, {
+    errorMap
+  });
+
+  if (!parsedConfig.success) {
+    throw new Error(parsedConfig.error.issues.map(i => i.message).join('\n'));
```
Comment (delucis) [inline packages/starlight/index.ts:28]: What do you think of prepending a line explaining that the error is in the `starlight` integration config?

Based on the example you shared, I’m imagining something like

```
15:23:05 [astro] Unable to load /home/otterlord/Documents/Repos/starlight/docs/astro.config.mjs

 error   Invalid config passed to starlight integration
  **logo**: Did not match union:
  > **logo.replacesTitle**: Expected type `"boolean"`, received "string"
```

**Review by delucis — COMMENTED** [review]
This looks great! Thanks so much for investigating @TheOtterlord.

Made one small suggestion for a possible improvement, but this is already a massive improvement on what we had before.

File: packages/starlight/index.ts (hunk 1 above)
Comment (TheOtterlord) [reply packages/starlight/index.ts]: Yeah, that sounds good :+1:

**Review by delucis — APPROVED** [review]
Perfect! The logging is pretty noisy because of how Vite handles the thrown error. Not sure what would be the fix there but this is already a huge improvement. Thanks again @TheOtterlord!

**Pull request merged**